src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.cs

```diff
@@ -189,4 +78,29 @@ protected static SyntaxTriviaList CreateCommentTrivia(SyntaxGenerator generator,
 
         return new SyntaxTriviaList(trivia);
     }
+
+    public async Task<Document> ImplementInterfaceAsync(
+        Document document,
+        IImplementInterfaceInfo info,
+        ImplementInterfaceOptions options,
+        ImplementTypeGenerationOptions generationOptions,
+        CancellationToken cancellationToken)
+    {
+        if (options.ImplementDisposePattern)
+        {
+            var generator = new ImplementInterfaceWithDisposePatternGenerator(
```

Comment:
still deferring to the existing 'generator' types that do some work and have some virtual behavior.  i don't like them, but i didn't want to churn too much.